test/evp_kdf_test.c

```diff
@@ -1678,6 +1678,132 @@ static int test_kdf_ss_kmac(void)
     return ret;
 }
 
+static 
+OSSL_PARAM *construct_snmpkdf_params(char *digest, unsigned char *eid, size_t eid_len,
+                                      unsigned char *password, size_t password_len)
+
+{
+    OSSL_PARAM *p = OPENSSL_malloc(sizeof(OSSL_PARAM) * 4);
+    OSSL_PARAM *p_ret = p;
+
+    if (!TEST_ptr(p))
+        return 0;
+
+    *p++ = OSSL_PARAM_construct_utf8_string(
+        OSSL_KDF_PARAM_DIGEST, digest, 0);
+    *p++ = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_SNMPKDF_EID, eid, eid_len);
+    *p++ = OSSL_PARAM_construct_octet_string(
+        OSSL_KDF_PARAM_PASSWORD, password, password_len);
+    *p = OSSL_PARAM_construct_end();
+
+    return p_ret;
+}
+
+static int test_kdf_snmpkdf_null_output_buf(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+
+    static unsigned char EID1[] = {
+          0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    unsigned char *password1 = NULL;
+
+    password1 = OPENSSL_malloc(1048577);
+    if (!TEST_ptr(password1))
+        return 0;
+
+    params = construct_snmpkdf_params("sha1", EID1, sizeof(EID1),
+                                      password1, 1048577);
+
+    /* Negative test - derive should fail */
+    ret = TEST_ptr(params)
+        && TEST_ptr(kctx = get_kdfbyname(OSSL_KDF_NAME_SNMPKDF))
+        && TEST_int_eq(EVP_KDF_derive(kctx, NULL, sizeof(out), NULL), 0);
+
+    EVP_KDF_CTX_free(kctx);
+    OPENSSL_free(password1);
+    OPENSSL_free(params);
+    return ret;
+}
+
+static int test_kdf_snmpkdf_bad_pw_len(void)
+{
+    int ret;
+    EVP_KDF_CTX *kctx = NULL;
+    OSSL_PARAM *params = NULL;
+    unsigned char out[SHA_DIGEST_LENGTH];
+
+    static unsigned char EID1[] = {
+          0x00, 0x00, 0x02, 0xb8, 0x77, 0x66, 0x55, 0x44, 0x33, 0x22, 0x11, 0x00 };
+    static unsigned char password1[] = {'L', 'P', 'U', 'Q', 'D', 'L', 's', 'K' };
+
+    params = construct_snmpkdf_params("sha1", EID1, sizeof(EID1),
+                                      (unsigned char *)password1, 1048577);
```

Comment:
Is this not the size of the password1 [] which is 8? (Otherwise it is reading out of bounds)

Reply:
This is a negative test for too long password. However to avoid UB please allocate an all-zero password of this length with OPENSSL_zalloc() - that should be fine for this test.

Reply:
Done.